### ISSUE TITLE
Chapter filter excludes extra chapters.

### ISSUE THREAD

**Issue by XellossMetallium** [Deathspike/mangarack#28]
This should be pretty trivial. The situation is the following: I have Sora no Otoshimono already archived by volume, so i used the command:
--chapter 68 http://bato.to/comic/_/comics/sora-no-otoshimono-r30
to exclude the volumes.
The chapters from 69 to 77 (last) were downloaded.
Yet, after 77, three more chapters were uploaded:
 Vol.14 Ch.Extra: Sora no Otoshimono: The Movie	
 Ch.Extra 2: Black and White!!
 Ch.Extra 1: The Strongest Middle Schooler
these files were skipped, probably because with the aforementioned filter active only numbered chapters are downloaded. 

Since saying that "I'm up to chapter 68" usually refers to the time ch68 is scanlated and not the chronological (story-wise) place of the chapter (meaning, I do want to download also an extra chapter for volume 3, if it is scanlated after the last chapter), I suppose this issue should be halfway between an enhancement and a bug.
If my idea is correct I don't know if this would require to do a check on the publishing date (in that case maybe this issue should be postponed after #2 ) or simply edit the rules to include different-named chapters (maybe without date check this would result in downloading also older extra chapters?).

Thank you as always!

**Comment by Deathspike** [comment]
This is an interesting issue. It's also quite a bit easier than #2 :+1:

I understand the confusion that comes from the --chapter and --volume filter, however, they are functioning exactly as intended and I do not wish to alter the functionality of these filters. It would break backwards compatibility and would make less sense intuitively. 

The appropriate way would be to introduce a new filter that focuses solely on publishing date. I imagine it would make a lot of sense to filter with something along the lines of `--since 20150418` (YYYYMMDD) to filter out anything that occurred before the provided date. Initial inspection on the 3 support providersshow these dates, albeit the implementation has to take care of funky notations such as '2 hours ago' and '3 weeks ago'. I have no idea how accurate the '3 weeks ago' is; is it EXACTLY 3 weeks, or SOMETHING IN 3 weeks? This might be a problem.

All in all, it seems like a great addition, but we'll need to be real careful of those date notations.

**Comment by XellossMetallium** [comment]
Indeed, a different filter is a good solution: more different option is better.

To clarify why I mentioned #2: 
Obviously implementing it properly would require a lot of work and options (to manage versions, scanlators etc, probably using a more complex "mpersistent" file), but.. allowing chapters after a certain date wouldn't download also newer versions of an older chapter? Empirically It shouldn't be so frequent for v2s, since they often came only hours after v1, but it could happen for chapters scanlated by a different group. Using the "since" filter in conjunction with "chapter" or "volume" obviously wouldn't solve the problem. 
In that sense, since #28 this is a relatively rare and not urgent occurrence, it could be best to postpone it after #2. Regardless of the time it will take solving the complex one should pose the bases to make the work on #28 a piece of cake.
As always this is from my less then limited experience in programming.

Now, regarding the actual issue:

About the "relative" time notations, the "since" filter could be by day at most, so "x minutes/hours/day ago" should be a go in all cases. 
"1-7 dayS ago" would require a conversion, but is accurate.
The problem would be "weeks": on the website (batoto) most certainly they are vague. 

Maybe the exact date is available somehow since after a month the reported date is the exact one.
I don't know if this could be a reliable proof but the app I use on android, called Mangawatcher (very nice app, by the way, if only comicrack could reach that level), always reports the exact date when the chapter is added. Obviously it could also be that the app can be so accurate because it checks batoto on a three hours basis (so the info is not "scraped", just "created"), this I cannot say for sure.. 

On chrome I tried to select "inspect element" on a picture from a chapter. The relative code is:

<img id="comic_page" style="max-width: 100%; width: 1089px;" src="http://img.bato.to/comics/2015/04/16/d/read5530119ae7be8/img000001.png" alt="Dungeon ni Deai wo Motomeru no wa Machigatteiru Darou ka - ch 33 Page 1 | Batoto!" onerror="this.src='http://img.bato.to/comics/misc/Img-Error.jpg'" onload="adjust_page_width();">

I don't know if it is useful but looks like the server stores the images in different folders by date (2015/04/16).

All things considered, shouldn't the info be available, the conversion could only be made approximative (on the "safe" side), like 1 week = 7 days, 2 weeks = 14 days.

**Comment by Deathspike** [comment]
Closed since `>= 4.0`. Filters are no longer supported.
